File: backend/src/modules/equipment/equipment.controller.js
const equipmentService = require('./equipment.service');
const siteVersionService = require('../siteVersions/siteVersion.service');

async function listByFloor(req, res) {
  const items = await equipmentService.listEquipmentByFloor(req.params.floorId);
  res.json(items);
}

async function createForFloor(req, res) {
  const equipment = await equipmentService.createEquipment(
    req.params.floorId,
    req.body
  );
  await siteVersionService.syncWorkingPayloadFromDb(equipment.siteId);
  res.status(201).json(equipment);
}

async function update(req, res) {
  const equipment = await equipmentService.updateEquipment(
    req.params.id,
    req.body
  );
  await siteVersionService.syncWorkingPayloadFromDb(equipment.siteId);
  res.json(equipment);
}

async function remove(req, res) {
  const existing = await equipmentService.deleteEquipment(req.params.id);
  await siteVersionService.syncWorkingPayloadFromDb(existing.siteId);
  res.status(204).send();
}

module.exports = {
  listByFloor,
  createForFloor,
  update,
  remove,
};

File: server/src/main/java/com/legioncontrols/server/sites/EquipmentController.java
package com.legioncontrols.server.sites;

import com.legioncontrols.server.releases.ReleaseService;
import java.util.List;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

/** Java equivalent of backend/src/modules/equipment/equipment.controller.js. */
@RestController
public class EquipmentController {

    private final EquipmentService equipmentService;
    private final ReleaseService releaseService;

    public EquipmentController(EquipmentService equipmentService, ReleaseService releaseService) {
        this.equipmentService = equipmentService;
        this.releaseService = releaseService;
    }

    @GetMapping("/api/floors/{floorId}/equipment")
    public List<EquipmentDto> listByFloor(@PathVariable String floorId) {
        return equipmentService.listEquipmentByFloor(floorId);
    }

    @GetMapping("/api/sites/{siteId}/equipment")
    public List<EquipmentDto> listBySite(@PathVariable String siteId) {
        return equipmentService.listEquipmentBySite(siteId);
    }

    @PostMapping("/api/floors/{floorId}/equipment")
    public ResponseEntity<EquipmentDto> createForFloor(@PathVariable String floorId, @RequestBody EquipmentService.CreateEquipmentRequest request) {
        var equipment = equipmentService.createEquipment(floorId, request);
        releaseService.syncWorkingPayloadFromDb(equipment.getSiteid());
        return ResponseEntity.status(HttpStatus.CREATED).body(EquipmentDto.from(equipment));
    }

    @PatchMapping("/api/equipment/{id}")
    public EquipmentDto update(@PathVariable String id, @RequestBody EquipmentService.UpdateEquipmentRequest request) {
        var equipment = equipmentService.updateEquipment(id, request);
        releaseService.syncWorkingPayloadFromDb(equipment.getSiteid());
        return EquipmentDto.from(equipment);
    }

    @DeleteMapping("/api/equipment/{id}")
    public ResponseEntity<Void> remove(@PathVariable String id) {
        var deleted = equipmentService.deleteEquipment(id);
        releaseService.syncWorkingPayloadFromDb(deleted.getSiteid());
        return ResponseEntity.noContent().build();
    }
}
